node_encoders.go

```diff
@@ -0,0 +1,120 @@
+package smt
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+// TODO_TECHDEBT: All of the parsing, encoding and checking functions in this file
+// can be abstracted out into the `trieNode` interface.
+
+// TODO_IMPROVE: We should create well-defined types & structs for every type of node
```

Comment:
Protobuf encoding and decoding could potentially slow down the tree - keeping with go types would be my preffered way of doing things IMO. Gob encoding is optimised for go types anyway